src/pybind/mgr/volumes/fs/volume.py

```diff
@@ -497,6 +497,34 @@ def remove_user_metadata(self, **kwargs):
                 ret = self.volume_exception_to_retval(ve)
         return ret
 
+    def regenerate_subvolume_metadata(self, **kwargs):
+        ret        = 0, "", ""
+        volname    = kwargs['vol_name']
+        subvolname = kwargs['sub_name']
+        groupname  = kwargs['group_name']
+        confirm    = kwargs['confirm']
+
+        if not confirm:
+            return (
+                -errno.EPERM, "",
+                "WARNING: This is a potentially disruptive operation and can clutter "
+                "a perfectly accessible subvolume. Add --yes-i-really-mean-it "
+                "if you are sure you wish to continue.")
+        try:
+            with open_volume(self, volname) as fs_handle:
+                with open_group(fs_handle, self.volspec, groupname) as group:
+                    try:
+                        with open_subvol(self.mgr, fs_handle, self.volspec, group, subvolname, SubvolumeOpType.REGENERATE) as subvolume:
```

Comment:
I think, this treats the subvolume as legacy and generates a legacy subvolume metadata file when the actual/original metadata file is truncated. So the metadata_mgr's config is pointing to _legacy config file and the config file you are generating is written to legacy config file instead of v2 config file in /volumes/<group>/<subvol>/.meta ?

Reply:
Yup, I am currently working on fixing that part only.